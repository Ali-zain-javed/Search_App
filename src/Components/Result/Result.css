#user {
    font-family: Arial, Helvetica, sans-serif;
    border-collapse: collapse;
    width: 100%;
  }

  .Outer{
      margin:60px
  }
  
  #user td, #user th {
    border: 1px solid #ddd;
    padding: 8px;
  }
  .error {
    border: 1px solid;
    margin: 10px 0px;
    padding: 15px 10px 15px 50px;
    background-repeat: no-repeat;
    background-position: 10px center;
    color: #D8000C;
    background-color: #FFBABA;
    background-image: url('https://i.imgur.com/GnyDvKN.png');
  }
 
 
  #user tr:nth-child(even){background-color: #f2f2f2;}
  
  #user tr:hover {background-color: #ddd;}
  
  #user th {
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: center;
    background-color: #04AA6D;
    color: white;
  }
  button {
    background-color: #04AA6D;
    color: white;
    padding: 14px 20px;
    margin: 8px 0;
    border: none;
    cursor: pointer;
    width: 100%;
  }