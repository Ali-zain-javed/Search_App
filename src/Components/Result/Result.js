import './Result.css';
import Pager from '../Pagination/Pagination.js'
function Result(props) {
    /**
  *In result component result show in table and pagination render 
  *
 */
    const { error = null, result = {}, SearchAgian = () => { } } = props;
    const { items = [], incomplete_results = true, total_count = 0 } = result || {}

    ///this method call on click page number in pagination then this function fetch new Items
    const fetchNewItem = (pageNumber) => {
        props.searchResult("", pageNumber)
    }
    return (
        <div className="Outer">
            {error && <div className="error">{error}</div>}
            <button className="button" style={{
                width: "144px",
                float: "right"
            }} onClick={SearchAgian}>Back to Search</button>
            <table id="user">
                <thead>
                    <tr>
                        <th>Image</th>
                        <th>User Name</th>
                        <th>Type</th>
                    </tr>
                </thead>
                <tbody>
                    {error == null && items && items.map((item, index) => <tr key={index}>
                        <td><img src={item.avatar_url} alt="loading..." width="100" height="70" /></td>
                        <td>{item.login}</td>
                        <td>{item.type}</td>
                    </tr>)}
                </tbody>
            </table>

            {items.length == 0 && <div style={{ textAlign: "center", marginTop: "20px" }}>No Record Found</div>}
            <div style={{ marginTop: "11px" }}>
                <Pager
                    currentPage={props.pageDetails && props.pageDetails.currentPage || 1}
                    pageLimit={props.pageDetails && props.pageDetails.limit || 30}
                    totalRecords={total_count}
                    onPageChanged={fetchNewItem || {}}
                />
            </div>
        </div>
    );
}

export default Result;
