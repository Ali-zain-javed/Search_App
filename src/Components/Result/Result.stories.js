import React from 'react';
import Result from './Result.js';
export default {
  component: Result,
  title: 'Components/Result',
};

//👇 We create a “template” of how args map to rendering
const Template = (args) => <Result {...args} />;

//👇 Each story then reuses that template
export const Primary = Template.bind({});

Primary.args = {
 result:{ items: [{ login: "ali", type: "user", avatar_url: "dsdd" },{ login: "ali", type: "user", avatar_url: "dsdd" }],
  incomplete_results: true,
  total_count: 1000},
  searchResult:(arg1,arg2)=>{console.log(arg1,arg2)}

};