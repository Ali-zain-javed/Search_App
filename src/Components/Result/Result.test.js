import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';
import { configure, mount, render, shallow } from 'enzyme';
import Result from './Result.js';
configure({ adapter: new Adapter() });
import ReactDOM from 'react-dom';
const matchSnapshot = (event) => { };

//moke data for testing
const data = { items: [{ login: "ali", type: "user", avatar_url: "dsdd" }], incomplete_results: true, total_count: 1000 }


describe('MyComponent', () => {
    it('In my test case should render correctly in "debug" mode', () => {
        const component = shallow(
            <Result error={null} result={[]} SearchAgian={() => { }} />
        );
        expect(component).toMatchSnapshot();
    });
});
it('In my test case should run result component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Result error={null} result={[]} SearchAgian={() => { }} />
        ,
        div
    );
    ReactDOM.unmountComponentAtNode(div);
});
it('In my test case should run result componeent with error props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Result error={"data not found"} result={[]} SearchAgian={() => { }} />
        ,
        div
    );
    ReactDOM.unmountComponentAtNode(div);
});
it('In my test case should run result componeent with result props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Result error={null} result={data} SearchAgian={() => { }} />
        ,
        div
    );
    ReactDOM.unmountComponentAtNode(div);
});
it('In my test case to be defind of table', () => {
    const wrapper = shallow(<Result searchResult={() => { }} error={null} />);
    expect(wrapper.find('.table')).toBeDefined();
});

it('in my test case table header are defined with length of three with name mentioned', () => {
    const wrapper = mount(
        <Result searchResult={() => { }} result={data} error={null} />
    );
    const rows = wrapper.find('table')
    expect(rows.length).toEqual(1)

    const firstRowColumns = rows.first().find('th').map(column => column.text())
    expect(firstRowColumns.length).toEqual(3)// since you have 4 td
    expect(firstRowColumns[0]).toEqual('Image')
    expect(firstRowColumns[1]).toEqual('User Name')
    expect(firstRowColumns[2]).toEqual('Type')
    expect(rows.find('th')).toHaveLength(3);
    //expect(rows.find('td')).toBeDefined(true);
});
it('In my test case my form should exist label field', () => {
    const wrapper = mount(
        <Result searchResult={() => { }} error={null} />
    );
    expect(wrapper.text()).toContain('Image');
    expect(wrapper.text()).toContain('User Name');
    expect(wrapper.text()).toContain('Type');
    expect(wrapper.text()).toContain('No Record Found');
});

