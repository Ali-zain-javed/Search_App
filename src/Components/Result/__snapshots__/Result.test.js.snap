// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent In my test case should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
